.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,55 @@
+name: release-workflow
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: Release to NPM
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Install pnpm
+        uses: pnpm/action-setup@v2.2.4
+        with:
+          version: 8
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.17
+          check-latest: true
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Check typescript
+        run: pnpm type:check
+
+      - name: Check eslint
+        run: pnpm lint
+
+      - name: Test by jest
+        run: pnpm test
```

Comment:
Isn't it already verified in `ci-workflow.yml` before pushing to main branch

**I don't think a double check is necessary** because this is not a part where human error can occur, is there a reason to verify it when you push it to the main one more time?

I think it's a waste of resources from the github action runner😊

Reply:
I don't think the only way to create a new commit on the main branch is through a Pull Request.

For example, even if it's not a pull request, I think we should check the quality at least once if there's a new commit in the main branch, either by force push or by other means. Are there any other good ways to satisfy this requirement?

Reply:
I didn't think of that, but I understood! Thank you.

I think there are two ways.

1. Protect the main branch with a [protect branch](https://kotlinworld.com/292) and limit direct push (We can take more stability) In this case, you don't have to do `ci` again.

2. If you are in a situation where you concerned (Push directly through main) I think we need this `test phrase`.

I think it would be good if the code that goes into main branch only be generated through Pull Request for review and history tracking in any situation, so I think number 1 would be good! How about you💬

Reply:
I turned protect branch rule on from now. and I remove 3 steps(Check typescript, Check eslint, Test by jest) in release-workflow.yml